### PR TITLE
3.x - Helidon DbClient JSON support

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
Added support to mark database as case insensitive in config file. This is required to run tests in both MySQL and Oracle DB with some shared code, because MySQL uses lower case identifiers and Oracle DB upper case by default.
JSON support verified against 21c in OCI:
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.helidon.tests.integration.dbclient.jdbc.oradb.ora21c.OraDbJsonSupportIT
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 32.964 s - in io.helidon.tests.integration.dbclient.jdbc.oradb.ora21c.OraDbJsonSupportIT
[INFO] Running io.helidon.tests.integration.dbclient.jdbc.oradb.tests.JsonDateTimeIT
[INFO] Tests run: 9, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.494 s - in io.helidon.tests.integration.dbclient.jdbc.oradb.tests.JsonDateTimeIT
[INFO] Running io.helidon.tests.integration.dbclient.jdbc.oradb.tests.JsonSupportIT
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.137 s - in io.helidon.tests.integration.dbclient.jdbc.oradb.tests.JsonSupportIT
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 20, Failures: 0, Errors: 0, Skipped: 0
```
And PostgreSQL/MySQL:
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.helidon.tests.integration.dbclient.jdbc.oradb.tests.JsonDateTimeIT
[INFO] Tests run: 9, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.435 s - in io.helidon.tests.integration.dbclient.jdbc.oradb.tests.JsonDateTimeIT
[INFO] Running io.helidon.tests.integration.dbclient.jdbc.oradb.tests.JsonSupportIT
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.043 s - in io.helidon.tests.integration.dbclient.jdbc.oradb.tests.JsonSupportIT
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 13, Failures: 0, Errors: 0, Skipped: 0
```

### CODE REVIEW

File: integrations/db/pgsql/src/main/resources/META-INF/services/io.helidon.common.mapper.spi.MapperProvider [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#
+# Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved.
```
Comment (tomas-langer) [inline integrations/db/pgsql/src/main/resources/META-INF/services/io.helidon.common.mapper.spi.MapperProvider:2]: There should be no service files in Helidon. Only declare the service in `module-info.java`

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: integrations/db/ojdbc/pom.xml [hunk 2]
```diff
@@ -25,7 +25,7 @@
         <version>2.3.2-SNAPSHOT</version>
     </parent>
 
-    <artifactId>ojdbc</artifactId>
+    <artifactId>helidon-integrations-db-ojdbc</artifactId>
```
Comment (barchetta) [inline integrations/db/ojdbc/pom.xml:28]: This looks like an incompatible change  (I also see you had to change the examples).  Is the artifact rename something that should wait until 3.0?

**Review by barchetta — CHANGES_REQUESTED** [review]

File: integrations/db/ojdbc/pom.xml (hunk 2 above)
Comment (Tomas-Kraus) [reply integrations/db/ojdbc/pom.xml:28]: Romain asked for those renames, because they do not follow naming conventions.

**Comment by Tomas-Kraus** [comment]
Anyway, this PR is blocked by Oracle JDBC driver release so there is no reason to rush. I'll fix notes from Tomas L. and we can discuss whether to change artifactId now or later.

File: dependencies/pom.xml [hunk 3]
```diff
@@ -52,7 +52,7 @@
         <version.lib.google-api-client>1.34.1</version.lib.google-api-client>
         <version.lib.google-oauth-client>1.33.3</version.lib.google-oauth-client>
         <version.lib.google-error-prone>2.3.3</version.lib.google-error-prone>
-        <version.lib.google-protobuf>3.21.4</version.lib.google-protobuf>
+        <version.lib.google-protobuf>3.20.1</version.lib.google-protobuf>
```
Comment (Tomas-Kraus) [inline dependencies/pom.xml:55]: Restore to 3.21.4!

**Comment by tomas-langer** [comment]
I am closing this PR, as it is not being worked on. Please @Tomas-Kraus create a new PR when you resolve the comments and rebase on latest.

**Pull request closed without merging**